app/models/concerns/pageflow/hosted_file.rb

```diff
@@ -4,63 +4,56 @@ module HostedFile
     include UploadedFile
 
     included do
-      has_attached_file(:attachment_on_filesystem, Pageflow.config.paperclip_filesystem_default_options)
-      has_attached_file(:attachment_on_s3, Pageflow.config.paperclip_s3_default_options)
+      alias_attribute :file_name, :attachment_file_name
+
+      has_attached_file(:attachment,
+                        Pageflow.config.paperclip_s3_default_options
+                          .merge(
+                            default_url: lambda do |attachment|
+                              attachment.instance.attachment_default_url
+                            end,
+                            styles: lambda do |attachment|
+                              attachment.instance.attachment_styles(attachment)
+                            end
+                          ))
```

Comment:
True. I changed the attachment name back to `attachment_on_s3` to avoid legacy problems. `ImageFile` (which is now a `HostedFile` as well) provides a new method `attachments_path_name` which is called in the Paperclip initializer for path interpolation and returns `processed_attachments`. For all other attachments this defaults to `attachment_on_s3s`